tests/Doctrine/Tests/ORM/Functional/SQLFilterTest.php

```diff
@@ -221,31 +221,38 @@ public function testEntityManagerIsFilterEnabled(): void
         self::assertFalse($em->getFilters()->isEnabled('foo_filter'));
     }
 
-    protected function configureFilters($em): void
+    private function configureFilters(EntityManagerInterface $em): void
     {
         // Add filters to the configuration of the EM
         $config = $em->getConfiguration();
         $config->addFilter('locale', '\Doctrine\Tests\ORM\Functional\MyLocaleFilter');
         $config->addFilter('soft_delete', '\Doctrine\Tests\ORM\Functional\MySoftDeleteFilter');
     }
 
-    protected function getMockConnection(): Connection
+    /**
+     * @return Connection&MockObject
+     */
+    private function getMockConnection(): Connection
     {
-        // Setup connection mock
         return $this->createMock(Connection::class);
     }
 
-    protected function getMockEntityManager(): EntityManagerInterface
+    /**
+     * @return EntityManagerInterface&MockObject
+     */
+    private function getMockEntityManager(): EntityManagerInterface
     {
-        // Setup entity manager mock
-        return $this->createMock(EntityManager::class);
+        return $this->createMock(EntityManagerInterface::class);
     }
 
-    protected function addMockFilterCollection(EntityManagerInterface $em): FilterCollection
+    /**
+     * @psalm-param EntityManagerInterface&MockObject $em
```

Comment:
Note: I'm using `@psalm-param` instead of `@param` because CodeSniffer will do strange this otherwise, see squizlabs/PHP_CodeSniffer#2887.